### ISSUE TITLE
Problems on Issues encountered during I2V rcm-training.

### ISSUE THREAD

**Issue by ariyali** [NVlabs/rcm#25]
Thank you very much for providing this framework!

We attempted RCM-based I2V distillation using the teacher model Wan2.1-Fun-V1.1-1.3B-Control-Camera, where the control signals include only the additional y and clip_feature. However, we observed that the distilled student model produces inference results with varying degrees of random blue-colored patches.

During training, we set the warmup steps to 1,000. Notably, these blue artifacts already started appearing in intermediate validation results at around step 1,500, and showed no significant improvement even after training up to 10,000 steps. Moreover, the blue patches are more pronounced when using inference with the reg parameter.

We also experimented with the sigma_max parameter during inference: When sigma_max = 80, the blue patches are relatively smaller, covering approximately 30% of the video area on average,  and when sigma_max = 5000, most of the video turns into solid blue.

We would greatly appreciate your insights on potential causes of this issue and whether any special adjustments or optimizations are needed during training.

For reference, we have verified that the teacher model (Wan2.1-Fun-V1.1-1.3B-Control-Camera) performs normal I2V inference under the same conditions: With sigma_max = 5000, it successfully generates coherent I2V results without camera-control signals. But with sigma_max = 80, however, the teacher output becomes highly random and fails to preserve the first-frame information or temporal consistency.

Thank you again for your help!

**Comment by wzr0108** [comment]
We are training Wan-Animate using only the sCM loss (without DMD), with `net_fake_score=false`, and we still observe noticeable blue artifacts in the generated videos.

https://github.com/user-attachments/assets/c5b56343-b933-460e-930d-babd20211d74

https://github.com/user-attachments/assets/7998fe26-43ce-4cc4-8f91-90dd2457e499

**Comment by zhengkw18** [comment]
Hi @wzr0108, maybe you can also try pure DMD

**Comment by ariyali** [comment]
We tried training using only DMD, adjusting parameters such as weights and noise distribution, but no matter how we tuned them, the image would turn blue around one thousand steps. However, when training with only SCM, the problem of turning blue does not occur.



---- Replied Message ----
| From | Kaiwen ***@***.***> |
| Date | 03/18/2026 22:05 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [NVlabs/rcm] Problems on Issues encountered during I2V rcm-training. (Issue #25) |
zhengkw18 left a comment (NVlabs/rcm#25)

Hi @wzr0108, maybe you can also try pure DMD

**Comment by zhengkw18** [comment]
> We tried training using only DMD, adjusting parameters such as weights and noise distribution, but no matter how we tuned them, the image would turn blue around one thousand steps. However, when training with only SCM, the problem of turning blue does not occur.
> […](#)

That is different observation to @wzr0108. For I2V, I recommend replacing the first frame of all denoised results to the ground-truth frame. This masks all losses on the first (condition) frame.

**Comment by ariyali** [comment]
We actually tried directly masking the loss of the first frame, but this only delayed the step at which the video turns blue—it didn't prevent the blue shift entirely. The issue mentioned by wzr0108—where training with only SCM still results in a blue shift—was also encountered by us after adjusting the noise sampling distribution of SCM. However, when training SCM with default parameters, we did not observe the blue shift.



---- Replied Message ----
| From | Kaiwen ***@***.***> |
| Date | 03/19/2026 05:26 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [NVlabs/rcm] Problems on Issues encountered during I2V rcm-training. (Issue #25) |
zhengkw18 left a comment (NVlabs/rcm#25)

We tried training using only DMD, adjusting parameters such as weights and noise distribution, but no matter how we tuned them, the image would turn blue around one thousand steps. However, when training with only SCM, the problem of turning blue does not occur.
…

That is different observation to @wzr0108. For I2V, I recommend replacing the first frame of all denoised results to the ground-truth frame. This masks all losses on the first frames.

**Comment by wzr0108** [comment]
Are these the default parameters?
```
p_G = L(LogNormal)(p_mean=0.7, p_std=1.6),
p_D = L(UniformShift)(shift=5.0),
max_simulation_steps_fake = 4,
state_t = 20,
sigma_max = 80,
```

**Comment by wzr0108** [comment]
> We actually tried directly masking the loss of the first frame, but this only delayed the step at which the video turns blue—it didn't prevent the blue shift entirely. The issue mentioned by wzr0108—where training with only SCM still results in a blue shift—was also encountered by us after adjusting the noise sampling distribution of SCM. However, when training SCM with default parameters, we did not observe the blue shift.
> […](#)

Did you use gradient accumulation?

**Comment by ariyali** [comment]
We do not use the gradient accumulation and it is the default parameter we use for 1.3B training



---- Replied Message ----
| From | Zhaorun ***@***.***> |
| Date | 03/19/2026 14:21 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [NVlabs/rcm] Problems on Issues encountered during I2V rcm-training. (Issue #25) |
wzr0108 left a comment (NVlabs/rcm#25)

We actually tried directly masking the loss of the first frame, but this only delayed the step at which the video turns blue—it didn't prevent the blue shift entirely. The issue mentioned by wzr0108—where training with only SCM still results in a blue shift—was also encountered by us after adjusting the noise sampling distribution of SCM. However, when training SCM with default parameters, we did not observe the blue shift.
…

Did you use gradient accumulation?

**Comment by zhengkw18** [comment]
I have added a discrete-time CM implementation https://github.com/NVlabs/rcm/blob/main/rcm/models/t2v_model_distill_rcm.py#L591, feel free to try it out by setting `model.config.cm_type=dcm`. In some cases, discrete-time CM performs more stably than sCM.
